lib/invoice_item_repository.rb

```diff
@@ -0,0 +1,58 @@
+require 'csv'
+
+class InvoiceItemRepository
+  def initialize(path)
+    @path = path
+    @rows = CSV.read(@path, headers: true, header_converters: :symbol)
+    @all = all
+  end
+
+  def all
+    result = @rows.map do |row|
+      InvoiceItem.new(row)
+    end
+  end
+
+  def find_by_id(id)
+    result = @all.find do |row|
+      row.id == id
+    end
+    result
+  end
+
+  def find_all_by_item_id(item_id)
+    result = @all.find_all do |row|
+      row.item_id == item_id
+    end
+    result
+  end
+
+  def find_all_by_invoice_id(invoice_id)
+    result = @all.find_all do |row|
+      row.invoice_id == invoice_id
+    end
+    result
+  end
+
+  def create(attributes)
+    attributes[:id] = @all.last.id.to_i + 1
+    @all << InvoiceItem.new(attributes)
+  end
+
+  def update(id, attributes)
+    invoice_item_to_update = find_by_id(id)
+    if attributes[:quantity] != nil
+      invoice_item_to_update.quantity = attributes[:quantity]
+    end
+    if attributes[:unit_price] != nil
+      invoice_item_to_update.unit_price = attributes[:unit_price]
+    end
```

Comment:
nice catch checking for nil values here!

I handled this using an **inline conditional** to eliminate a few lines of code, but it also made my lines really long. 
Inline conditionals would look like this:

`invoice_item_to_update.quantity = attributes[:quantity] if attributes[:quantity]`

The `if` essentially checks if the value exists or if it's `nil`, and then assigns the value to the variable if it it returns `truthy` (i.e. if it exists). If your lines get too long you can rename `invoice_item_to_update` and/or assign `attributes[:quantity]` to a variable, etc.